votingsystem/src/voting.rs

```diff
@@ -0,0 +1,280 @@
+        if transaction.tx_type() != "Vote" {
+            return Err(ExecuteError::InvalidMetadata)
+        }
+        let tx: OwnTransaction =
+            serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+        tx.verify().map_err(|_| ExecuteError::InvalidSign)?;
+
+        match tx.tx.action {
+            TxAction::CreateVotePaper(TxCreateVotePaper {
+                general_meeting_id,
+                agenda_number,
+                authorization_token,
+                voter_public_key,
+            }) => {
+                let vote_paper =
+                    VotePaper::new(general_meeting_id.clone(), agenda_number, authorization_token, voter_public_key);
+                self.storage_mut().set(vote_paper.vote_paper_id.0.as_ref(), serde_cbor::to_vec(&vote_paper).unwrap());
+                Ok(vote_paper.vote_paper_id)
+            }
+
+            TxAction::Vote(Ballot {
+                vote_paper_id,
+                choice,
+                voter_signature,
+            }) => {
+                let mut vote_paper: VotePaper =
+                    serde_cbor::from_slice(&self.storage().get(vote_paper_id.0.as_ref()).unwrap()).unwrap();
+                let ballot = Vote::new(choice, voter_signature);
+                let vote_id = ballot.clone().ballot_id;
+                vote_paper.set_used_in(vote_id.clone());
+
+                let meeting_id = vote_paper.get_general_meeting_id();
+                self.storage_mut().set(vote_id.as_ref(), serde_cbor::to_vec(&ballot).unwrap());
+                self.storage_mut().set(meeting_id.as_ref(), serde_cbor::to_vec(&ballot).unwrap());
+                Ok(vote_id.id)
+            }
+        }
+    }
+}
+
+impl Stateful for Context {
+    fn set_storage(&mut self, storage: ServiceRef<dyn SubStorageAccess>) {
+        self.storage.replace(storage.unwrap_import().into_remote());
+    }
+}
+
+impl TxOwner for Context {
+    fn block_opened(&mut self, _: &Header) -> Result<(), HeaderError> {
+        Ok(())
+    }
+
+    fn execute_transaction(&mut self, transaction: &Transaction) -> Result<TransactionOutcome, ()> {
+        if let Err(error) = self.excute_tx(transaction) {
+            match error {
+                ExecuteError::InvalidMetadata => Err(()),
+                ExecuteError::InvalidSign => Err(()),
+                ExecuteError::InvalidFormat => Err(()),
+                ExecuteError::InvalidAdmin => Err(()),
+                ExecuteError::UsedVotePaper => Err(()),
+            }
+        } else {
+            Ok(Default::default())
+        }
+    }
+
+    fn check_transaction(&self, transaction: &Transaction) -> Result<(), coordinator::types::ErrorCode> {
+        let todo_fixthis: coordinator::types::ErrorCode = 3;
+        assert_eq!(transaction.tx_type(), "Vote");
+        let tx: OwnTransaction = serde_cbor::from_slice(&transaction.body()).map_err(|_| todo_fixthis)?;
+        tx.verify().map_err(|_| todo_fixthis)?;
+        Ok(())
```

Comment:
Please check these things.

TxAction::CreateVotePaper:

* general meeting is exist or not
* general meeting is finished or not
* voting time is finished or not
* agenda number exceeds the agenda count in the general meeting or not

TxAction::Vote

* vote paper is already used or not
* choice should be one of 4 choices
* voter signature is generated from voter public key or not